src/App.tsx

```diff
@@ -17,18 +46,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                onQueryChange={setQuery}
+                status={todoStatus}
+                onStatusChange={setTodoStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length !== 0 ? (
```

Comment:
you can leave it as is. Or if you want to prevent such a moment, you can introduce an additional _isLoading_ state variable
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/54936044/3f159fa2-0065-43a0-bbab-1e679bb69fe3)